pkg/pgmodel/sql_ingest.go

```diff
@@ -579,6 +584,20 @@ func tryRecovery(conn pgxConn, req copyRequest, err error) error {
 		return err
 	}
 
+	// if there was an epoch abort, notify the requester that they should update
+	// their cache
+	if pgErr.Code == "PS001" {
+		// notify the requester that they should update their cache.
+		// this _must never block_: since it is a back edge blocking will cause
+		// deadlocks
+		// TODO should we also return the data so it INSERT can be retried?
```

Comment:
I'd remove this TODO. I don't see it on even our longer-term roadmap since Prometheus itself retries. And this should happen ~never